modules/articles/client/articles.client.module.js

```diff
@@ -1,7 +1,7 @@
 (function (app) {
   'use strict';
 
-  app.registerModule('articles');
+  app.registerModule('articles', ['core']);
```

Comment:
I like your suggestion..something like "Core module is required for special route handling"

I am not sure either about the module loading order, from the [docs](https://docs.angularjs.org/guide/module) it describes how modules are asynchronously loaded:
>Because modules do nothing at load time they can be loaded into the VM in any order and thus script loaders can take advantage of this property and parallelize the loading process.

but then, the docs describe how dependencies are loaded first:

> Modules can list other modules as their dependencies. Depending on a module implies that the required module needs to be loaded before the requiring module is loaded. In other words the configuration blocks of the required modules execute before the configuration blocks of the requiring module.